File: lib-integration-tests/src/test/java/io/github/kory33/guardedqueries/core/testcases/parser/TestCaseParsers.java
package io.github.kory33.guardedqueries.core.testcases.parser;

import uk.ac.ox.cs.gsat.GTGD;
import uk.ac.ox.cs.pdq.fol.ConjunctiveQuery;
import uk.ac.ox.cs.pdq.fol.TypedConstant;

public class TestCaseParsers {
    public static final FormulaParsingContext commonParsingContext = new FormulaParsingContext(
            // we regard all symbols of the form "c_{number}" as constants
            s -> s.startsWith("c_") && s.length() > 2 && s.substring(2).chars().allMatch(Character::isDigit),
            TypedConstant::create
    );

    public static final FormulaParsers.WhitespaceIgnoringParser<ConjunctiveQuery> conjunctiveQuery =
            new FormulaParsers.WhitespaceIgnoringParser<>(FormulaParsers.conjunctiveQueryParser(commonParsingContext));

    public static final FormulaParsers.WhitespaceIgnoringParser<GTGD> gtgd =
            new FormulaParsers.WhitespaceIgnoringParser<>(FormulaParsers.gtgdParser(commonParsingContext));
}

File: core-integration-tests/src/test/scala/io/github/kory33/guardedqueries/core/testcases/TestCaseParsers.scala
package io.github.kory33.guardedqueries.core.testcases

import io.github.kory33.guardedqueries.parser.{FormulaParsers, FormulaParsingContext}
import uk.ac.ox.cs.gsat.GTGD
import uk.ac.ox.cs.pdq.fol.{ConjunctiveQuery, TypedConstant}

object TestCaseParsers {
  val commonParsingContext: FormulaParsingContext =
    FormulaParsingContext(
      // we regard all symbols of the form "c_{number}" as constants
      (s: String) =>
        s.startsWith("c_") && s.length > 2 && s.substring(2).chars.allMatch(Character.isDigit),
      TypedConstant.create
    )

  val conjunctiveQuery: FormulaParsers.IgnoreWhitespaces[ConjunctiveQuery] =
    FormulaParsers.IgnoreWhitespaces(
      FormulaParsers.conjunctiveQueryParser(using commonParsingContext)
    )

  val gtgd: FormulaParsers.IgnoreWhitespaces[GTGD] =
    FormulaParsers.IgnoreWhitespaces(FormulaParsers.gtgdParser(using commonParsingContext))
}
